src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,8 +289,49 @@ def algorithm(self, parameters=None):
             with CodeTimer("Rucio block resolution", logger=logging):
                 self.findRucioBlocks(uniqueActivePU, msPileupList)
 
-            # TODO in future tickets: find the json files in the spec/sandbox area and tweak them
+            # define where to look for sandbox tar balls
+            idir = getattr(self.config.WorkflowUpdater, "componentDir")
+            if not os.path.exists(idir):
+                idir = '/data/srv/wmagent/current/install/wmagentpy3'
+            if not os.path.exists(idir):
+                self.logger.error("Directory %s does not exist", idir)
+                return
+
+            # loop over existing pileup names and
+            # find the json files in the spec/sandbox area and tweak them
+            for rec in msPileupList:
```

Comment:
This block would be better organized in its own method as well, instead of expanding it all in the algorithm() method. There are multiple ways to do so, either passing the pileup records to the method, or calling the method for each pileup. 

In any case, make sure to have try/except blocks protecting the code, to avoid skipping the whole cycle in case something (soft) unexpected happens.

Reply:
I put this block into separate function, but did not put in there try/except since I call this function within tr/except block and I doubt we need exception within exception handling. Please review and let me know

Reply:
ok, fine, I moved idir into constrcutor.